Docs: Updates override docs and related docs (#1867)
vtex/faststore: apps/site/pages/docs/building-sections/overriding-components-and-props.mdx

---
title: '  Overriding native components and props'
sidebar_label: 'Component Customization'
---

import { Callout, Steps, Tab, Tabs } from 'nextra-theme-docs'

<header>
# Overriding Native Components and Props

</header>

<Callout type="warning" emoji="⚠️">
  This documentation is currently being developed.
</Callout>

In this guide, you will learn how to modify a component's behavior using additional properties in your store code and create new components by overriding their native equivalents. 

These features are useful when you need a customized behavior for a specific component not available in the [FastStore native options](/docs/reference-guides/list-of-native-sections).

<Callout type="warning" emoji="⚠️">

>   When overriding a component, it's important to note that you will lose the integration developed in [`@faststore/core`](/docs/getting-started/3-project-structure#faststorecore) for that component. To avoid compromising the existing integration, we highly recommend to consider [overriding props](#1-override-native-component-by-overriding-its-props) or utilizing the [theming tools](/docs/customizing/components) provided by `@faststore/core` to achieve the desired behavior. For more information about the available props for each component, refer to the [Components UI](/components) guide.

Author: ```suggestion
We highly recommend considering [overriding props](#1-override-native-component-by-overriding-its-props) or utilizing the [theming tools](/docs/customizing/components) provided by `@faststore/core` to achieve the desired behavior. Overriding a component inside a section will compromise the existing integration developed in [`@faststore/core`](/docs/getting-started/3-project-structure#faststorecore) and lose updates for that component.
```

---
title: '  Overriding native components and props'
sidebar_label: 'Component Customization'
---

import { Callout, Steps, Tab, Tabs } from 'nextra-theme-docs'

<header>
# Overriding Native Components and Props

</header>

<Callout type="warning" emoji="⚠️">
  This documentation is currently being developed.
</Callout>

In this guide, you will learn how to modify a component's behavior using additional properties in your store code and create new components by overriding their native equivalents.

These features are useful when you need a customized behavior for a specific component not available in the [FastStore native options](/docs/reference-guides/list-of-native-sections).

> <Callout type="warning" emoji="⚠️">
>   When overriding a component, it's important to note that you will lose the
>   integration developed in
>   [`@faststore/core`](/docs/getting-started/3-project-structure#faststorecore)
>   for that component. To avoid compromising the existing integration, we highly
>   recommend to consider [overriding
>   props](#1-override-native-component-by-overriding-its-props) or utilizing the
>   [theming tools](/docs/customizing/components) provided by `@faststore/core` to
>   achieve the desired behavior. For more information about the available props
>   for each component, refer to the [Components UI](/components) guide.

Author: ```suggestion
We highly recommend considering [overriding props](#override-native-components-props) or utilizing the [theming tools](/docs/customizing/components) provided by `@faststore/core` to achieve the desired behavior. Overriding a component inside a section will compromise the existing integration developed in [`@faststore/core`](/docs/getting-started/3-project-structure#faststorecore) by losing updates for that component.
```

![overrides-PDP-buy-button-component-alert](https://vtexhelp.vtexassets.com/assets/docs/src/overrides-buy-button-open-alert___b62f8cb5f3f98f6bd0f5502081de6a9b.gif)

## Extra: Overriding components and passing props

In certain scenarios, you may have the option to use prior integrations within your custom component.
Building upon our previous example, we could pass the `BuyButton` props to our new component.
This approach allows us to incorporate the icon and cart sidebar integration while also introducing additional functionality to the `onClick` function, such as triggering an alert or sending an event to analytics.

```tsx filename="CustomBuyButton.tsx"
import React from 'react'
import { Button as UIButton, ButtonProps } from '@faststore/ui'

export function CustomBuyButton(props: ButtonProps) {
  return (
    <UIButton
      {...props}
      variant="primary"
      onClick={(event) => {
        alert('You will love this product!')
        props.onClick?.(event)
      }}
    >

>       My New Buy Button

Author: ```suggestion
      New Buy Button
```
